File: Projects/Projects_vb/ZengFanyun/cmds_Monitoring.vb
Imports System
Imports Autodesk.Revit.UI
Imports Autodesk.Revit.DB
Imports OldW.Instrumentation
Imports OldW.Soil
Imports rvtTools_ez.ExtensionMethods
Imports OldW.DataManager
Imports OldW.Excavation
Imports Autodesk.Revit.DB.Events

Namespace OldW.Commands

    <Autodesk.Revit.Attributes.Transaction(Autodesk.Revit.Attributes.TransactionMode.Manual)>
    Public Class cmd_DataEdit : Implements IExternalCommand

        Public Function Execute(commandData As ExternalCommandData, ByRef message As String, elements As ElementSet) As Result Implements IExternalCommand.Execute
            Dim uiApp As UIApplication = commandData.Application
            Dim doc As Document = uiApp.ActiveUIDocument.Document
            Dim eleIds As ICollection(Of ElementId) = uiApp.ActiveUIDocument.Selection.GetElementIds

            Dim frm As New ElementDataManager(eleIds, doc)
            frm.ShowDialog()
            Return Result.Succeeded
        End Function

    End Class

End Namespace

File: Projects/Projects/ZengFanyun/cmds_Monitoring.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using Autodesk.Revit.DB;
using Autodesk.Revit.UI;

//using Autodesk.Revit.DB;
//using Autodesk.Revit.UI;
// End of VB project level imports


namespace OldW.Commands
{
	
	[Autodesk.Revit.Attributes.Transaction(Autodesk.Revit.Attributes.TransactionMode.Manual)]public class cmd_DataEdit : IExternalCommand
		{
		
		public Result Execute(ExternalCommandData commandData, ref string message, ElementSet elements)
		{
			UIApplication uiApp = commandData.Application;
			Document doc = uiApp.ActiveUIDocument.Document;
			ICollection<ElementId> eleIds = uiApp.ActiveUIDocument.Selection.GetElementIds();
			
			OldW.DataManager.ElementDataManager frm = new OldW.DataManager.ElementDataManager(eleIds, doc);
			frm.ShowDialog();
			return Result.Succeeded;
		}
		
	}
	
}

